workflow/Snakefile

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python3
+
+import pandas as pd
+import os
+import time
+global files
+
+configfile: "config/config.yaml"
+
+FILES = glob_wildcards(config["fastq_folder"] + "/{sample}.fastq.gz")    # get all fastq files
+NAMES = FILES.sample    # extract sample names
+
+REFERENCE = glob_wildcards("reference_genomes/{ref}.fa")    # get all reference genomes 
+REF = REFERENCE.ref    # extract reference genome names
+
+if config["now_time"]:
+    config["output"] = f'{config["now_time"]}_{config["output"]}'
+
+rule all:    # specify the final output files
+    input:
+        expand(f'{config["output"]}/fastp/report/{{sample}}.html', sample=NAMES),    # fastp report
+        expand(f'{config["output"]}/fastp/{{sample}}_filt.fastq.gz', sample=NAMES),    # filtered reads
+        f'{config["output"]}/samtools/minimum_error_rates.csv',    # minimum error rates [bc**,ref_*,*.****]
+        expand(f'{config["output"]}/samtools/{{sample}}.{{ref}}.bam', sample=NAMES, ref=REF),    # sorted bam files
+        expand(f'{config["output"]}/consensus/{{sample}}.{{ref}}.fa', sample=NAMES, ref=REF),    # consensus sequences
+        expand(f'{config["output"]}/consensus/medaka/{{sample}}.{{ref}}', sample=NAMES, ref=REF),    # medaka consensus sequences
+        expand(f'{config["output"]}/freebayes/{{sample}}.{{ref}}.vcf', sample=NAMES, ref=REF),    # variant calling
+        f'{config["output"]}/samtools/view_stats.csv',    # total mapped reads
+        f'{config["output"]}/samtools/view_stats_rt.csv',    # mapped reads of the RT region
+        f'{config["output"]}/copy_files_done.txt',   # files copied to output folder
+        f'{config["output"]}/vcf_modifications_done.txt',    # vcf files modified
+        f'{config["output"]}/qc.csv'    # quality control file
+
+
+# Preprocess the fastq files
+rule fastp:
+    input:
+        fastq = f'{config["fastq_folder"]}/{{sample}}.fastq.gz'    # get all fastq files
+    output: 
+        fastq = f'{config["output"]}/fastp/{{sample}}_filt.fastq.gz',    # output filtered reads
+        report = f'{config["output"]}/fastp/report/{{sample}}.html'
+    singularity:
+        config["fastp"]
```

Comment:
You mentioned snakemake8 bringing some special functionality running singularities. This looks very standard?